lib/rouge/lexers/hlsl.rb

```diff
@@ -0,0 +1,166 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    load_lexer 'c.rb'
+
+    class HLSL < C
+      title "HLSL"
+      desc "HLSL, the High Level Shading Language for DirectX (docs.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl)"
+      tag 'hlsl'
+      filenames '*.hlsl', '*.hlsli'
+      mimetypes 'text/x-hlsl'
+
+      def self.keywords
+        @keywords ||= Set.new %w(
+          asm asm_fragment break case cbuffer centroid class column_major
+          compile compile_fragment const continue default discard do else export
+          extern for fxgroup globallycoherent groupshared if in inline inout
+          interface line lineadj linear namespace nointerpolation noperspective
+          NULL out packoffset pass pixelfragment point precise return register
+          row_major sample sampler shared stateblock stateblock_state static
+          struct switch tbuffer technique technique10 technique11 texture
+          typedef triangle uniform vertexfragment volatile while
+        )
+      end
+
+      def self.keywords_type
+        @keywords_type ||= Set.new %w(
+          dword matrix snorm string unorm unsigned void vector BlendState Buffer
+          ByteAddressBuffer ComputeShader DepthStencilState DepthStencilView
+          DomainShader GeometryShader HullShader InputPatch LineStream
+          OutputPatch PixelShader PointStream RasterizerState RenderTargetView
+          RasterizerOrderedBuffer RasterizerOrderedByteAddressBuffer
+          RasterizerOrderedStructuredBuffer RasterizerOrderedTexture1D
+          RasterizerOrderedTexture1DArray RasterizerOrderedTexture2D
+          RasterizerOrderedTexture2DArray RasterizerOrderedTexture3D RWBuffer
+          RWByteAddressBuffer RWStructuredBuffer RWTexture1D RWTexture1DArray
+          RWTexture2D RWTexture2DArray RWTexture3D SamplerState
+          SamplerComparisonState StructuredBuffer Texture1D Texture1DArray
+          Texture2D Texture2DArray Texture2DMS Texture2DMSArray Texture3D
+          TextureCube TextureCubeArray TriangleStream VertexShader
+
```

Comment:
@mitjmcc Unfortunately you can't include comments in Ruby's string array literals. I kept the logical groupings (albeit wrapping at 80 characters).